homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -92,45 +82,97 @@ def make_zmanim(date):
-        today_times = make_zmanim(today)
+        today_times = self.make_zmanim(today)
 
         if now > sunset:
             after_shkia_date = date.next_day
 
         if today_times.havdalah and now > today_times.havdalah:
             after_tzais_date = date.next_day
 
+        self._state = self.get_state(after_shkia_date, after_tzais_date)
+        _LOGGER.debug("New value for %s: %s", self._type, self._state)
+
+    def make_zmanim(self, date):
+        """Create a Zmanim object."""
+        return hdate.Zmanim(
+            date=date,
+            location=self._location,
+            candle_lighting_offset=self._candle_lighting_offset,
+            havdalah_offset=self._havdalah_offset,
+            hebrew=self._hebrew,
+        )
+
+    def get_state(self, after_shkia_date, after_tzais_date):
+        """For a given type of sensor, return the state."""
         # Terminology note: by convention in py-libhdate library, "upcoming"
         # refers to "current" or "upcoming" dates.
         if self._type == "date":
-            self._state = after_shkia_date.hebrew_date
-        elif self._type == "weekly_portion":
+            return after_shkia_date.hebrew_date
+        if self._type == "weekly_portion":
             # Compute the weekly portion based on the upcoming shabbat.
-            self._state = after_tzais_date.upcoming_shabbat.parasha
-        elif self._type == "holiday_name":
-            self._state = after_shkia_date.holiday_description
-        elif self._type == "holiday_type":
-            self._state = after_shkia_date.holiday_type
-        elif self._type == "upcoming_shabbat_candle_lighting":
-            times = make_zmanim(after_tzais_date.upcoming_shabbat.previous_day.gdate)
-            self._state = times.candle_lighting
-        elif self._type == "upcoming_candle_lighting":
-            times = make_zmanim(
+            return after_tzais_date.upcoming_shabbat.parasha
+        if self._type == "holiday_name":
+            return after_shkia_date.holiday_description
+        if self._type == "holiday_type":
+            return after_shkia_date.holiday_type
+        if self._type == "omer_count":
+            return after_shkia_date.omer_day
+
+        return None
+
+
+class JewishCalendarTimeSensor(JewishCalendarSensor):
+    """Implement attrbutes for sensors returning times."""
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return dt_util.as_utc(self._state) if self._state is not None else None
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        return "timestamp"
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {}
+
+        if self._state is None:
+            return attrs
+
+        attrs["year"] = self._state.year
+        attrs["month"] = self._state.month
+        attrs["day"] = self._state.day
+        attrs["hour"] = self._state.hour
```

Comment:
Do these times also need changing to reflect utc time?

Reply:
I removed them, don't see the value where one would just want the hour of one of these times.
Also, I think, if implemented, these should go in the sensor component for the timestamp device class.